buildpipeline/Core-Setup-RHEL7-x64.json

```diff
@@ -20,55 +20,74 @@
     },
     {
       "enabled": true,
-      "continueOnError": false,
-      "alwaysRun": false,
-      "displayName": "Docker cleanup",
+      "continueOnError": true,
+      "alwaysRun": true,
+      "displayName": "Initialize tools",
       "timeoutInMinutes": 0,
       "task": {
-        "id": "d9bafed4-0b18-4f58-968d-86655b4d2ce9",
-        "versionSpec": "1.*",
+        "id": "6c731c3c-3c68-459a-a5c9-bde6e6595b5b",
+        "versionSpec": "2.*",
         "definitionType": "task"
       },
       "inputs": {
-        "filename": "bash",
-        "arguments": "-c \"docker ps -a -q --filter name=$(DockerContainerName) | xargs --no-run-if-empty docker rm -f -v\"",
-        "workingFolder": "",
+        "scriptPath": "pkg/init-tools.sh",
+        "args": "",
+        "disableAutoCwd": "false",
+        "cwd": "",
         "failOnStandardError": "false"
       }
     },
     {
       "enabled": true,
       "continueOnError": false,
       "alwaysRun": false,
-      "displayName": "Run build in Docker container",
+      "displayName": "Initialize Docker",
       "timeoutInMinutes": 0,
       "task": {
         "id": "6c731c3c-3c68-459a-a5c9-bde6e6595b5b",
         "versionSpec": "2.*",
         "definitionType": "task"
       },
       "inputs": {
-        "scriptPath": "scripts/dockerrun-as-current-user.sh",
-        "args": "-t --rm --sig-proxy=true --name $(DockerContainerName) -v $(Build.SourcesDirectory):/opt/code -w /opt/code -e CONNECTION_STRING -e PUBLISH_TO_AZURE_BLOB microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2 /bin/bash -c \"HOME=/opt/code; git clean -X -d -f; ./build.sh $(BuildArguments)\"",
+        "scriptPath": "pkg/Tools/scripts/docker/init-docker.sh",
+        "args": "$(DockerImageName)",
         "disableAutoCwd": "false",
         "cwd": "",
         "failOnStandardError": "false"
       }
     },
     {
       "enabled": true,
-      "continueOnError": true,
-      "alwaysRun": true,
-      "displayName": "Initialize tools",
+      "continueOnError": false,
+      "alwaysRun": false,
+      "displayName": "Docker cleanup",
```

Comment:
There is no need for this step with the introduction of init-docker.  It cleans up all containers.